contracts/token/ERC1155/Tatum1155.sol

```diff
@@ -0,0 +1,739 @@
+
+        for (uint256 i = 0; i < accounts.length; ++i) {
+            batchBalances[i] = balanceOf(accounts[i], ids[i]);
+        }
+
+        return batchBalances;
+    }
+
+    /**
+     * @dev See {IERC1155-setApprovalForAll}.
+     */
+    function setApprovalForAll(address operator, bool approved)
+        public
+        virtual
+        override
+    {
+        require(
+            _msgSender() != operator,
+            "ERC1155: setting approval status for self"
+        );
+
+        _operatorApprovals[_msgSender()][operator] = approved;
+        emit ApprovalForAll(_msgSender(), operator, approved);
+    }
+
+    /**
+     * @dev See {IERC1155-isApprovedForAll}.
+     */
+    function isApprovedForAll(address account, address operator)
+        public
+        view
+        virtual
+        override
+        returns (bool)
+    {
+        return _operatorApprovals[account][operator];
+    }
+
+    /**
+     * @dev See {IERC1155-safeTransferCashback}.
+     */
+    function safeTransferCashback(
+        address from,
+        address to,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) public payable {
+        require(to != address(0), "ERC1155: transfer to the zero address");
+        require(from == _msgSender(), "ERC1155: caller is not owner");
+
+        address operator = _msgSender();
+
+        _beforeTokenTransfer(
+            operator,
+            from,
+            to,
+            _asSingletonArray(id),
+            _asSingletonArray(amount),
+            data
+        );
+
+        uint256 fromBalance = _balances[id][from];
+        require(
+            fromBalance >= amount,
+            "ERC1155: insufficient balance for transfer"
+        );
+        _balances[id][from] = fromBalance - amount;
+        _balances[id][to] += amount;
+        if (_cashbackRecipients[id].length != 0) {
+            uint256 sum = 0;
+            for (uint256 i = 0; i < _cashbackValues[id].length; i++) {
+                sum += _cashbackValues[id][i];
+            }
+            require(
+                msg.value > sum,
+                "ERC1155: value must be greater than cashback values"
+            );
+            for (uint256 i = 0; i < _cashbackRecipients[id].length; i++) {
+                // transferring cashback to authors
+                payable(_cashbackRecipients[id][i]).transfer(
+                    _cashbackValues[id][i]
+                );
+            }
+            if (msg.value > sum) {
+                payable(msg.sender).transfer(msg.value - sum);
+            }
+        } else {
+            if (msg.value > 0) {
+                payable(msg.sender).transfer(msg.value);
+            }
+        }
+        emit TransferSingle(operator, from, to, id, amount);
+
+        _doSafeTransferAcceptanceCheck(operator, from, to, id, amount, data);
+    }
+
+    /**
+     * @dev See {IERC1155-safeBatchTransferFrom}.
+     */
+    function safeBatchTransferCashback(
```

Comment:
cant we merge it together inside safeBatchTransfer?

Reply:
please do so

Reply:
This is from previous commit, already done